File: Java/MaxConsecutiveOnes/MaxConsecutiveOnes.java
/**
 * (10ms)
 * 这个方法很简单，就是遍历整个数组，在遍历的同时更新最长1的个数。一旦遇到0，count值就复位。
 * 时间复杂度为O(n),空间复杂度为O(1)
 * @author Phoenix-Z
 *
 */
public class MaxConsecutiveOnes {

	public int findMaxConsecutiveOnes(int[] nums) {
		int count = 0;
		int maxCount = count;
		for(int num : nums) {
			if(num == 1) {
				count++;
				maxCount = count > maxCount ? count : maxCount;
			} else {
				count = 0;
			}
		}
		return maxCount;
    }
	
}

File: Python/MaxConsecutiveOnes/MaxConsecutiveOnes.py
# coding=utf-8
# 由于输入只有0和1，所以可以考虑先将数组转化为字符串，再根据0进行分割，最后再找出最长的
# 连续的是1的字符串。显然这个方法不是最高的，但是只需要一行代码。
class Solution(object):
	def findMaxConsecutiveOnes(self, nums):
		"""
		:type nums: List[int]
		:rtype: int
		"""
		# (112ms)
		return max([s.count('1') for s in ''.join(map(str, nums)).split('0')])
		# 下面这个代码更加简洁
		# (106ms)
		# return max(map(len, ''.join(map(str, nums)).split('0')))
		
print Solution().findMaxConsecutiveOnes([1,1,0,0,1,1,1])